Code/Numerics/Vector.h

```diff
@@ -22,6 +22,8 @@
 #include <boost/random.hpp>
 #include <boost/smart_ptr.hpp>
 
+static constexpr double zero_tolerance = 1.e-6;
```

Comment:
This is, I think, too big. I'd prefer to go with something like 1e-16

Reply:
Yeah, that wold be closer to an actual zero, but I wonder if normalizing vectors with length under 1e-6 does make sense. E.g. in AddHs we use 1e.-4 for the square distance, which should be even smaller.

Reply:
> Yeah, that wold be closer to an actual zero, but I wonder if normalizing vectors with length under 1e-6 does make sense. E.g. in AddHs we use 1e.-4 for the square distance, which should be even smaller.

Without knowing the exact context, I don't think there's a good general answer for a lower bound on what makes sense. Going with something that's quite close to zero seems like the safest thing to do.